src/app/navigation/share-account-table/share-account-table.component.ts

```diff
@@ -0,0 +1,65 @@
+/** Angular Imports */
+import { Component, OnInit, Input, ViewChild } from '@angular/core';
+import { MatTableDataSource, MatPaginator, MatSort } from '@angular/material';
+
+/** Custom Pipes */
+import { AccountsFilterPipe } from '../../pipes/accounts-filter.pipe';
+
+@Component({
+  selector: 'mifosx-share-account-table',
+  templateUrl: './share-account-table.component.html',
+  styleUrls: ['./share-account-table.component.scss']
+})
+export class ShareAccountTableComponent implements OnInit {
+
+  /** Columns to be displayed in the share accounts table. */
+  displayedColumns: string[] = ['accountNo', 'productName', 'totalApprovedShares', 'Status'];
+  /** Show closed share accounts */
+  showClosed = false;
+  /** Data source for share account table. */
+  dataSource: MatTableDataSource<any>;
+  /** Share Account Data */
+  accountData: any;
+
+  /** Paginator for share account table. */
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  /** Sorter for share account table. */
+  @ViewChild(MatSort) sort: MatSort;
+
+  /** Share Account Setter */
+  @Input() set shareAccountData(data: any) {
+    this.accountData = data;
+    const filteredAccountData = this.accountsFilterPipe.transform(data, 'share', this.showClosed ? 'closed' : 'open');
```

Comment:
Doesn't really matter what 'open' is, it could be 'Germany' and it would work fine. The if statement just checks whether showClosed is set as 'closed' or not, but cannot be empty.